hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -519,6 +519,20 @@ def all(cls) -> List["HealthCheck"]:
     This check requires the :ref:`Hypothesis pytest plugin<pytest-plugin>`,
     which is enabled by default when running Hypothesis inside pytest."""
 
+    differing_test_runners = 10
+    """Checks if :func:`@given <hypothesis.given>` has been applied to a test
+    which is executed by different test runners. If the test runners have different
+    test behaviour, this leads to flaky or nonreproducible tests when replaying from
+    the database.
+
+    The correct fix is often to bring the test runner instance under the control of
+    hypothesis by explicit parametrization over or sampling from subclasses, or to
+    refactor so that :func:`@given <hypothesis.given>` is specified on leaf subclasses.
+
+    Suppress this health check only if the different runners are known to behave
+    identically during tests; alternatively, disable database access or implement "
+    "__eq__ on the test runner class."""
```

Comment:
I'd delete this whole paragraph.   I don't expect users to reliably implement the right `__eq__` semantics, and would thus use `is` rather than `==` internally.  I also prefer to never recommend disabling the database, again because I think most people who would follow that advice would be better off if they didn't.